### PR TITLE
Add skeleton for NearestNeighborTerm in C++.

### PR DESCRIPTION
**Pull request by geirst**
@arnej27959 please review
@havardpe FYI

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/matching/termdatafromnode.cpp [hunk 1]
```diff
@@ -42,6 +42,7 @@ struct TermDataFromTermVisitor
     void visit(ProtonSuffixTerm &n) override { visitTerm(n); }
     void visit(ProtonPredicateQuery &) override { }
     void visit(ProtonRegExpTerm &n) override { visitTerm(n); }
+    void visit(ProtonNearestNeighborTerm &) override { }
```
Comment (havardpe) [inline searchcore/src/vespa/searchcore/proton/matching/termdatafromnode.cpp]: if it is a proton term, you should be able to extract the term concept from it

File: searchcorespi/src/vespa/searchcorespi/index/indexcollection.cpp [hunk 2]
```diff
@@ -203,6 +202,7 @@ class CreateBlueprintVisitor : public search::query::QueryVisitor {
     void visit(SuffixTerm &n)      override { visitTerm(n); }
     void visit(PredicateQuery &n)  override { visitTerm(n); }
     void visit(RegExpTerm &n)      override { visitTerm(n); }
+    void visit(NearestNeighborTerm &) override { }
```
Comment (havardpe) [inline searchcorespi/src/vespa/searchcorespi/index/indexcollection.cpp]: consider adding TODO

**Comment by geirst** [comment]
@havardpe PTAL

**Comment by havardpe** [comment]
👍

**Pull request merged**